### PR TITLE
Disable debugging encrypted realms

### LINKED ISSUE #1625: LLDB hangs when debugging an application that is using an encrypted Realm.
LLDB hangs in Xcode 6.2 when stepping (or continuing) through code in a project that has accessed an encrypted Realm in any way. The hang occurs even when you are not stepping over Realm statements. This is very easy to reproduce and happens every time, thus making debugging applications using encrypted Realms impossible.

I have also sent you a project to reproduce this issue with.

### PR DESCRIPTION
**Pull request by bdash**
Prevent debugging of an application using an encrypted Realm to work around
frequent LLDB hangs. Until the underlying issue is addressed you may set
REALM_DISABLE_ENCRYPTION=YES in your application's environment variables to
have requests to open an encrypted Realm treated as a request for an
unencrypted Realm.

Fixes #1625.

/cc @jpsim @tgoyne @segiddins

### CODE REVIEW

**Comment by tgoyne** [comment]
It might be better to skip the encryption tests entirely if a debugger is attached. I've had to disable them quite a few times to be able to debug tests which are run after them. CI will still run them, and if a developer needs to run them locally they can comment out that check.

**Comment by bdash** [comment]
Xcode appears to always run the tests under debugserver, even when using "xcodebuild test" like CI does.

File: Realm/RLMRealm.h [hunk 1]
```diff
@@ -99,6 +99,13 @@
  The on-disk storage for encrypted Realms are encrypted using AES256+HMAC-SHA2,
  but otherwise they behave like normal persisted Realms.
 
+ Encrypted Realms currently cannot be opened while lldb is attached to the
```
Comment (tgoyne) [inline Realm/RLMRealm.h]: This bit isn't true anymore (it was true when we used a signal handler), and should instead say something about debugging with encrypted realms leading to lldb hanging. A link to the bug discussing would be nice too.

**Comment by bdash** [comment]
@tgoyne I revised the changes to address the issues you mentioned.

**Comment by tgoyne** [comment]
:+1:

**Pull request merged**